presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RemoveRedundantAggregateDistinct.java

```diff
@@ -38,17 +40,12 @@
         implements Rule<AggregationNode>
 {
     private static final Pattern<AggregationNode> PATTERN = aggregation()
-            .matching(RemoveRedundantAggregateDistinct::aggregateDistinctOfUniqueKey);
+            .matching(RemoveRedundantAggregateDistinct::hasAggregations);
 
-    private static boolean aggregateDistinctOfUniqueKey(AggregationNode node)
+    private static boolean hasAggregations(AggregationNode node)
```

Comment:
why was the distinct check moved out of the pattern?

Reply:
It is because of the comment by rongrong at https://github.com/prestodb/presto/pull/16416#discussion_r893003401. Pasting it here "Since you have to run this same logic again in apply, might as well use a simpler pattern, and check for match once here."